### ISSUE TITLE
error messages don't print the full path

### ISSUE THREAD

**Issue by vikstrous** [docker/distribution#1793]
Example extremely confusing error message:
```
{"level":"fatal","msg":"failed to garbage collect: error enumerating blobs: filesystem: Path not found: /docker/registry/v2/blobs","time":"2016-06-16T22:16:04.133505465Z"}
```

It turns out that the path printed here is relative, not absolute and it's relative to the path configured for the storage driver. We need to either remove the leading `/` or include the full path.

**Comment by amouat** [comment]
I had a similar error message where the path was just "/". That confused me for longer than I'm willing to admit.

**Comment by stevvooe** [comment]
@vikstrous @amouat I am confused. What change are you proposing exactly? This looks like an absolute path. It is possible that GC was run on an empty registry?

**Comment by amouat** [comment]
It wasn't an absolute path, it was relative to `/var/lib/registry`. Basically, I created my own image for arm and it took me a while to realise the error message meant I had to do `mkdir /var/lib/registry`.

**Comment by stevvooe** [comment]
@amouat So, there error messages are mapped onto paths as seen by the storage driver. Not all storage drivers have a path prefix, so having a path there would be incorrect in most cases, such as with s3 or azure or drivers that map a path space directly.

Given that information what change do you see us making? Would something in the documentation help?

**Comment by amouat** [comment]
Right, so the problem is the error message assumes a bit much knowledge on part of the user. Consider changing to a message such as:

> "Storage path X not found. Note that this path is relative to the root of the storage driver."


Even better:

> "Storage path X not found. Note that this path is relative to the root of the filesystem storage driver which currently points to /var/lib/registry"

Or:

> "Storage path X not found. Note that this path is relative to the root of the AWS storage driver which currently points to ..."

**Comment by stevvooe** [comment]
@amouat Yes, we assume that the user has basic operating knowledge of the system and doesn't make assumptions about their level of operating knowledge of the system.

If you are willing to submit a PR that does this, feel free, but this looks like a documentation issue.

**Comment by stevvooe** [comment]
@amouat A good patch here would put the configuration value for `rootdirectory` on the affect log messages. That allow storage drivers that have a root directory to declare so without generating a massive about bytes in the log output.

**Comment by chimeworld** [comment]
I'd like to work on this if no one else is.
